### PR TITLE
Clean up crio-status interface with additional output options

### PR DESCRIPTION
**Pull request by fire833**
Signed-off by: Kendall Tauser <kttpsy@gmail.com>

#### Does this PR introduce a user-facing change?

```release-note

Updates crio-status with a -o option to specify alternate output formats for the different data objects that can be returned.

```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Hey @fire833, thank you for this PR. May I ask you to sign your commits with `git commit --amend -s`?

**Comment by fire833** [comment]
Sorry for delay, done.

**Comment by saschagrunert** [comment]
@fire833 unfortunately all commits have to be signed, can you squash them together please?

**Comment by fire833** [comment]
I just rebased back to the original commit. Sorry about this I am a bit new to PRs in OSS. Let me know if this works.

File: cmd/crio-status/main.go [hunk 1]
```diff
@@ -106,29 +118,12 @@ func containers(c *cli.Context) error {
...
-	for k, v := range info.Labels {
-		fmt.Printf("  %s: %s\n", k, v)
-	}
-	fmt.Printf("annotations:\n")
-	for k, v := range info.Annotations {
-		fmt.Printf("  %s: %s\n", k, v)
+	if output, e := genOutput(info); e == nil {
+		fmt.Print(output)
+		return nil
+	} else {
```
Comment (haircommander) [inline cmd/crio-status/main.go:124]: nit, typically, if we process an error, we would use that block to do so, like:
```
output, err := genOutput(info)
if err != nil {
    return e
}
fmt.Println(output)
```

mostly a style thing, but I would prefer it if you wouldn't mind

File: cmd/crio-status/main.go [hunk 2]
```diff
@@ -142,20 +137,35 @@ func info(c *cli.Context) error {
...
+		return e
 	}
+}
 
-	return nil
+func genOutput(input types.Output) (string, error) {
+	switch outputArg {
+	case "text":
+		{
```
Comment (haircommander) [inline cmd/crio-status/main.go:151]: no need for the `{` (linter is saying the same)

**Comment by sohankunkerkar** [comment]
we don't need this change anymore as we are planning to deprecate `crio-status` in v1.28 (https://github.com/cri-o/cri-o/pull/6557)

**Pull request closed without merging**